### DISCUSSION TITLE
positive values for account regint seem unused

### DISCUSSION THREAD

**Discussion by bassdscho** [baresip/baresip#2463] [Q&A]
I am using a pure baresip v2.6.0 on arm / raspberry and did some wireshark traces from baresip to a minisipserver.

As I understand, the account / `regint` value shall be used as the registration interval when the registration to the registrar failed for whatever reason. Also, it is used to decide if a registration shall happen at all or if this account shall be considered a local / peer to peer account. After a successful registration, the `rwait` value comes into account.

However, the registration intervals seen with wireshark (when I deliberately use a non-existing account) seems to be determined only by the `failwait(...)` time from `libre/src/sipreg/reg.c`. At least, I don't really find where the `reg->expires` property is used in a `tmr_start(...)` call elsewhere in `libre/src/sipreg/reg.c`. I haven't live debugged the code yet however, this observation is just from reading through the source code. 

Registration packets from baresip are seen immediately after starting baresip and then in increasing, rather random intervals, just as what `failwait(...)` would do. I would expect a static interval instead. The `fbregint` interval is used in a `tmr_start(...)` call and works as expected.

Here's my account config line:

`<sip:*****@*****;transport=tcp>;answermode=auto;answerdelay=0;autelev_pt=101;sip_autoanswer=no;sip_autoanswer_beep=off;auth_user=*****;auth_pass=*****;call_transfer=no;fbregint=0;regint=42;prio=0;rwait=95;pubint=0`

Is this a bug or did I miss something?

Might be related: https://github.com/baresip/re/discussions/307

Edit: the regint param is used as expiration interval in the contact header of a register request (https://www.rfc-editor.org/rfc/rfc3261#section-10.2.1.1). So it's wrong by me to claim it is unused. 

From the accounts wiki I thought this might be some actively controllable period of the baresip UA:

> regint=n accounts with a positive n will create a User-Agent for the registration at a SIP-registrar. Thus the User-Agent starts to send SIP REGISTER packets periodically.

**Comment by cspiel1** [comment]
https://github.com/baresip/baresip/wiki/Accounts#regint

regint is the normal registration interval. It is not for the failed case. This is for the normal, positive case.

fbregint is used to "poll" fallback SIP servers. A fallback SIP servers is used for registering to, if the main SIP server is not available.
